sdks/python/apache_beam/runners/portability/abstract_job_service.py

```diff
@@ -278,6 +279,23 @@ def to_runner_api(self):
         state=self.state)
 
 
+class JarArtifactManager(object):
+  def __init__(self, jar_path, root):
+    self._root = root
+    self._zipfile_handle = zipfile.ZipFile(jar_path, 'a')
+
+  def close(self):
+    self._zipfile_handle.close()
+
+  def file_writer(self, path):
```

Comment:
Can we get a type annotation and/or comment on the return value here?

Reply:
Done.